sdk/containerregistry/Azure.Containers.ContainerRegistry/tests/ContainerRepositoryClientLiveTests.cs

```diff
@@ -109,7 +113,256 @@ public async Task CanSetRepositoryProperties()
+            {
+                Assert.GreaterOrEqual(page.Values.Count, pageSize);
+                pageCount++;
+            }
+
+            // Assert
+            Assert.IsTrue(pageCount >= minExpectedPages);
+        }
+
+        [RecordedTest]
+        public async Task CanGetArtifactsStartingMidCollection()
+        {
+            // Arrange
+            var client = CreateClient();
+            int pageSize = 1;
+            int minExpectedPages = 2;
+
+            AsyncPageable<RegistryArtifactProperties> artifacts = client.GetRegistryArtifactsAsync();
+            string firstDigest = null;
+            string secondDigest = null;
+            int artifactCount = 0;
+            await foreach (var artifact in artifacts)
+            {
+                if (artifactCount == 0)
+                {
+                    firstDigest = artifact.Digest;
+                }
+
+                if (artifactCount == 1)
+                {
+                    secondDigest = artifact.Digest;
+                    break;
+                }
+
+                artifactCount++;
+            }
+
+            // Act
+            artifacts = client.GetRegistryArtifactsAsync();
+            var pages = artifacts.AsPages($"</acr/v1/{_repositoryName}/_manifests?last={firstDigest}&n={pageSize}>");
+
+            int pageCount = 0;
+            Page<RegistryArtifactProperties> firstPage = null;
+            await foreach (var page in pages)
+            {
+                if (pageCount == 0)
+                {
+                    firstPage = page;
+                }
+
+                Assert.LessOrEqual(page.Values.Count, pageSize);
+                pageCount++;
+            }
+
+            // Assert
+            Assert.AreNotEqual(null, firstPage);
+            Assert.AreEqual(secondDigest, firstPage.Values[0].Digest);
+            Assert.GreaterOrEqual(pageCount, minExpectedPages);
+        }
+
+        [RecordedTest]
+        public async Task CanGetArtifactsOrdered()
+        {
+            // Arrange
+            string repository = $"library/node";
+            string tag = "newest";
+            var client = CreateClient(repository);
+            if (this.Mode != RecordedTestMode.Playback)
+            {
+                await ImportImage(repository, tag);
+            }
+
+            // Act
+            AsyncPageable<RegistryArtifactProperties> artifacts = client.GetRegistryArtifactsAsync(new GetRegistryArtifactOptions(RegistryArtifactOrderBy.LastUpdatedOnDescending));
+
+            // Assert
+            string digest = null;
+            await foreach (RegistryArtifactProperties artifact in artifacts)
+            {
+                digest = artifact.Digest;
+                Assert.That(artifact.Repository.Contains(repository));
+                Assert.That(artifact.Tags.Contains(tag));
+                break;
+            }
+
+            // Clean up
+            await client.DeleteRegistryArtifactAsync(digest);
```

Comment:
nit: You may want to consider putting this into a finally block if this test creates this artifact so that it still cleans up in the event of an Assert failure above.

Reply:
Good idea!  I'll add this.